lib/core/sidenav-layout/components/sidenav-layout/sidenav-layout.component.ts

```diff
@@ -56,7 +59,9 @@ export class SidenavLayoutComponent implements OnInit, AfterViewInit, OnDestroy
         isMenuMinimized: () => this.isMenuMinimized
     };
 
-    constructor(private mediaMatcher: MediaMatcher) {
+    constructor(private mediaMatcher: MediaMatcher,  private config: AppConfigService, private storage: StorageService) {
```

Comment:
The preservation should not be part of this component. The component should communicate the collapsed/opened state with an event bubbling up, and in the application, where we use this component we can do the saving to the local storage. For example in our case in AMA, everything would go through the store, and as a side effect we could  save to the localstorage.